runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/QueryResponse.java

```diff
@@ -0,0 +1,48 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import lombok.Getter;
+import org.corfudb.format.Types;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+import static org.corfudb.format.Types.OrchestratorResponseType.WORKFLOW_STATUS;
+
+/**
+ * Return a boolean that indicates whether a particular workflow is being executed by
+ * an orchestrator service.
```

Comment:
what about the status of it's actions . Maybe you return only final status, right ?

Reply:
Can't, we don't persist the history of actions. So once an action completes, it gets removed from memory.